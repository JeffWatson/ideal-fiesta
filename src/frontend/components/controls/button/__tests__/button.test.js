import React from 'react';
import { shallow } from 'enzyme';
import Button from '../button';

describe('frontend/components/button component', () => {
  const props = {
    className: 'class',
    text: 'text',
    onClick: jest.fn(),
    children: (<div>Children</div>),
  };

  it('renders correctly', () => {
    const button = shallow(<Button {...props} />);

    expect(button).toMatchSnapshot();
  });

  it('invokes onClick when clicked', () => {
    const button = shallow(<Button {...props} />);

    button.find('.button').simulate('click');
    expect(props.onClick.mock.calls.length).toEqual(1);
  });
});
