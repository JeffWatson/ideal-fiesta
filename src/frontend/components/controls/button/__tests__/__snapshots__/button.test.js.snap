// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontend/components/button component renders correctly 1`] = `
<button
  className="button class"
  disabled={false}
  onClick={[Function]}
>
  text
  <div>
    Children
  </div>
</button>
`;
